db/migrate/20220110085736_create_reservations.rb

```diff
@@ -0,0 +1,14 @@
+class CreateReservations < ActiveRecord::Migration[7.0]
+  def change
+    create_table :reservations do |t|
+      t.belongs_to :seance
+      t.belongs_to :ticketdesk
+      t.belongs_to :client
+      t.has_many :ticket
```

Comment:
`has_many` is not migration thing - it is just a way to tell rails that given model has specific 1 to many relation in database, but the foreign keys for this relation are on different table, this may help you understand how this works: https://www.sqlshack.com/learn-sql-types-of-relations/

Reply:
If I understand previous comment from @wmaciejak should I use like `add_reference(:seance, :movie, foreign_key: true)` and delete `t.has_many t.belong_to` from migrations ?

Reply:
`t.belong_to` is all right - [it is just an alias for `add_reference` ](https://edgeguides.rubyonrails.org/active_record_migrations.html#references

But there is no `has_many` function.
Migrations are about changes in the database, not about application logic.
So let's talk about one-to-many relation in the database, totally ignoring the rails app for now.
You create one-to-many relation between table `reservations` and table `tickets` by adding on `tickets` foreign key `reservation_id` - that way each ticket _belongs to_ one reservation, and reservation can have multiple tickets. We do not add anything on `reservations` table itself, the foreign key on  `tickets` table is enough to create that relation in database. 

So in migrations, there is no need to specify on `reservations` that it has many `tickets`.

Now - model and application logic is different. In Rails app we specify both on `Ticket` model that it `belongs_to :reservation` - to denote that there is foreign key `reservation_id` - and on `Reservation` model we specify that it `has_many  :tickets` - that  way when we use it, it will know that it can find related tickets by looking for all tickets with `reservation_id` equal to that reservation id.